source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -85,15 +102,44 @@ AWSLambdaConfigImpl::AWSLambdaConfigImpl(
   }
 }
 
+void AWSLambdaConfigImpl::init() {
+  // Add file watcher for token file
+  auto shared_this = shared_from_this();
+  file_watcher_->addWatch(
+      token_file_, Filesystem::Watcher::Events::Modified,
+      [shared_this](uint32_t) {
+        try {
+          const auto web_token = shared_this->api_.fileSystem().fileReadToEnd(
```

Comment:
is this the only place we read the token?
i think we should also do it in the config ctor, so that if the file does not exist, the config should get nack'ed.
wdyt?

Reply:
Talked on slack and decided to move the throws into the constructor for the `StsCredentialsProviderFactoryImpl`